File: AIペアプログラミング実践編_ソースコード/practical-python/session-4/Ans/inventory/models/product.py
from django.db import models


class ProductCategory(models.TextChoices):
    """商品カテゴリ"""
    STATIONERY = 'STATIONERY', '文房具'
    OFFICE = 'OFFICE', 'オフィス用品'
    OTHER = 'OTHER', 'その他'


class Product(models.Model):
    """商品マスタ"""
    name = models.CharField('商品名', max_length=100)
    category = models.CharField(
        'カテゴリ',
        max_length=20,
        choices=ProductCategory.choices,
        default=ProductCategory.OTHER
    )
    price = models.PositiveIntegerField('単価')
    created_at = models.DateTimeField('作成日時', auto_now_add=True)
    updated_at = models.DateTimeField('更新日時', auto_now=True)

    class Meta:
        db_table = 'products'
        verbose_name = '商品'
        verbose_name_plural = '商品'

    def __str__(self):
        return self.name

File: AIペアプログラミング実践編_ソースコード/practical-ts/session-2/Ans/src/models/product.ts
/**
 * 商品カテゴリ
 */
export enum ProductCategory {
  /** 文房具 */
  STATIONERY = 'STATIONERY',
  /** オフィス用品 */
  OFFICE = 'OFFICE',
  /** その他 */
  OTHER = 'OTHER',
}

/**
 * 商品マスタ
 */
export interface Product {
  /** 商品ID（自動採番） */
  id: number
  /** 商品名 */
  name: string
  /** カテゴリ */
  category: ProductCategory
  /** 単価（税抜） */
  price: number
  /** 作成日時 */
  createdAt: Date
  /** 更新日時 */
  updatedAt: Date
}

/**
 * 商品登録時の入力
 */
export interface ProductInput {
  /** 商品名 */
  name: string
  /** カテゴリ */
  category: ProductCategory
  /** 単価（税抜） */
  price: number
}

/**
 * 商品更新時の入力（部分更新可能）
 */
export interface ProductUpdateInput {
  /** 商品名 */
  name?: string
  /** カテゴリ */
  category?: ProductCategory
  /** 単価（税抜） */
  price?: number
}
